File: MaxDepthBTree.java
solution for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        return helper(root, 0, Integer.MIN_VALUE);
    }
    
    public int helper(TreeNode root, int count, int max) {
        if (root == null) {
            return Integer.max(count, max);
        }
        
        int leftcount = helper(root.left, count + 1, max);
        int rightcount = helper(root.right, count + 1, max);
        
        return Integer.max(leftcount, rightcount);
    }
}

File: MaxDepthBTree.cpp
tion for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        return helper(root, 0, INT_MIN);
    }
    
    int helper(TreeNode* root, int count, int max) {
        if (root == NULL) {
            return count > max ? count : max;
        }
        
        int leftcount = helper(root->left, count + 1, max);
        int rightcount = helper(root->right, count + 1, max);
        
        return leftcount > rightcount ? leftcount : rightcount;
    }
};
